feat: implement new landing page

Shaimaa01 opened the pull request:
### Ticket Numbers:-
FTVT3-32: Develop the page in React & TypeScript - add hero image - add image for core functionalities
FTVT3-30: Create copy for Landing Page sections - name of the app - one liner description
FTVT3-39: Create copy for the core functionalities: dashboard | savings tab (allow user to manage saving goals) | budget tab (allow user to manage budgets per spending category)

Shaimaa01 commented:
Please take a moment to review it and let me know if any adjustments or improvements are needed. Thank you!

jackli921 commented:
Landing page looks amazing! But it prompts questions
1. Where did the design come from?
2. How did you code the landing page up? I thought you had just finished learning React

jackli921 reviewed (approved):
The styling and animations techniques involved here is beyond my frontend knowledge, so I have not much to add besides spinning it up locally and seeing that it looks great. 

Really curious how the overall design, starry night animation, and all the styling was decided.

kayoMichael commented on apps/client/src/pages/Index.tsx:
```diff
@@ -0,0 +1,22 @@
+import Footer from '@/components/landing/Footer';
+import Hero from '@/components/landing/Hero';
+import Nav from '@/components/landing/Nav';
+
```
For future development, it would be helpful to organize pages into subdirectories within the pages directory. Since we'll likely have multiple pages as the app grows, creating a folder like pages/landing for the landing page will keep things cleaner and more maintainable.

Also, using a more descriptive filename such as LandingPage.tsx (instead of a generic index.tsx) improves clarity and makes it easier to navigate the codebase.

Shaimaa01 replied:
I did look at it. Is that what you meant? Is there another adjustment

Shaimaa01 committed:
fix: reorganize landing page into pages/landing/LandingPage.tsx

Shaimaa01 committed:
chore: remove package-lock.json from repo

Shaimaa01 committed:
feat: update Features and scoped LandingPage styles

- Updated content in the Features section
- Scoped LandingPage styles using CSS module
- Restored default shadcn/ui styles in index.css for global use

Shaimaa01 committed:
Resolve merge conflict in bun.lock by accepting development version

kayoMichael reviewed (approved):
LGTM!
